api/src/main/java/io/strimzi/api/kafka/model/connector/KafkaConnectorOffsetsAnnotation.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.connector;
+
+public enum KafkaConnectorOffsetsAnnotation {
```

Comment:
I guess I can accept that we have the annotations enum in the `api`. But it probably should have only the valid values and not `none` or `unknown`, or? Why is that needed?

Reply:
I added them to use when processing the annotation in the code, but in reality I'm only using `none` as the default when fetching the annotation so I'll remove `unknown`. I placed this class in `api` as I figured it's part of the API, but it's actually only used from the `cluster-operator` module, so should I actually move it there?

Reply:
I was wondering if it should be in the `api`. I'm personally not a big fan of it. But we already do have `KafkaRebalanceAnnotation` as an example of this being in the `api` module before. So I felt like I should not insist on this being moved (funny enough, it also includes the same options you have there which I did not notice yesterday 🙄). 

I think the logic of having it in the API is that users might use it to set the annotation programmatically. But then it would be better if it contained the real values that the user is expected to use and not the `none` and `unknown` values.

So, I don't know ... I would do it differently. But it does not feel fair to insist you change it if you are happy with it 🤷.

Reply:
That's a fair point Jakun instead. I was thinking about the same (while working on auto-rebalancing) but while i.e. refresh, approve, stop are needed by the users, the none and unknown are there just as internal values for the rebalance logic. Maybe Kate could address this here (now) and we should do the same on the rebalance operator side.

Reply:
@ppatierno I've removed `unknown` since I don't use that internally. When you say "Maybe Kate could address this here (now)" do you mean moving the class to `cluster-operator` module?

Reply:
`unknown` was removed but `none` is still there which is not a valid annotation, it's just the absence of it.
IIRC the idea of having these annotation in the api module was because of providing enums to a user who is developing an automation on handling a `KafkaRebalance` resource, so instead of writing a string, they can use our enum. Of course, having `none` (and `unknown`) brings confusion. So or we put these `none` somewhere else leaving the enum in the api module or we just remove it from the api module (and we should do the same for the rebalancing operator) leaving the user to use the api module but then just strings when it comes to apply these annotations.

Reply:
I've moved it to `io.strimzi.operator.cluster.operator.resource` in the `cluster-operator` module, if you're both happy with that I'll mark this thread as resolved.

Reply:
You mean the whole annotation class? Would `io.strimzi.operator.cluster.model` be better?

Reply:
Agree on the model package.